crates/router/src/core/payments/flows/session_flow.rs

```diff
@@ -247,13 +288,29 @@ async fn create_applepay_session_token(
             router_data.request.to_owned(),
         )?;
 
+        let required_billing_contact_fields = is_dynamic_fields_required(
+            state,
+            enums::PaymentMethod::Wallet,
+            enums::PaymentMethodType::ApplePay,
+            &connector.connector_name,
+            enums::FieldType::UserBillingName,
```

Comment:
We can't decide here based on one field right? We need to check if any of the billing details are required instead!